### ISSUE TITLE
WeasyPrint does not render iframe content

### ISSUE THREAD

**Issue by chutchut** [Kozea/WeasyPrint#1010]
I am unable to render iframe content with WeasyPrint, and see no references in the documentation (or many in the code) in regards to handling of iframes. This is similar to a previous issue (https://github.com/Kozea/WeasyPrint/issues/840) that was closed without resolution. 

I have attempted (https://github.com/Kozea/WeasyPrint/issues/840#issuecomment-478487626) to follow the guidance of the author by ensuring the iframe tag has appropriate style defining explicit height/width, and used a static document with no JS for both the src and srcdoc attributes of the iframe to no avail. Please provide example code or guidance in regard to rendering iframes. 

```
>>> print(cairo.cairo_version())
11600

pip freeze:
cairocffi==1.1.0
CairoSVG==2.4.2
cffi==1.13.2
cssselect2==0.2.2
defusedxml==0.6.0
html5lib==1.0.1
pbr==4.2.0
Pillow==6.2.1
pycparser==2.19
Pyphen==0.9.5
six==1.11.0
stevedore==1.29.0
tinycss2==1.0.2
virtualenv==16.0.0
virtualenv-clone==0.3.0
virtualenvwrapper==4.8.2
WeasyPrint==50
webencodings==0.5.1
```

Test HTML

```
<!doctype html>
<html>
<style>
body iframe {
   width: 720px;
   height: 720px;
}
</style>
<body>
<iframe srcdoc="<html><body><b>XXX</b></body></html>"><b>YYY</b></iframe>
</body>
</html>
```

Expected output (as rendered in browser):

![image](https://user-images.githubusercontent.com/6414463/71088394-32d74080-2196-11ea-97bf-83cfe5c59513.png)

Actual output (as rendered in PDF):

![image](https://user-images.githubusercontent.com/6414463/71088470-674afc80-2196-11ea-9e37-3e5c2c11cac5.png)

**Comment by liZe** [comment]
Hello @chutchut,

Thank you for this bug report!

WeasyPrint only knows `src` attribute for iframes, not `srcdoc`. That’s why it doesn’t work.

If you’re new and are interested to contribute: that’s a very good issue to dive into WeasyPrint’s code! Just ask, I can help you :smile:.

**Comment by chutchut** [comment]
Hi @liZe, thanks for the prompt response. I would be happy to help in terms of adding support for that attribute, but maybe not until after Xmas! :)

The reason to use the srcdoc attribute in the PoC was just to make it simpler, **but this issue also occurs when using the src attribute** as mentioned in the issue and demonstrated below. The src attribute in the PoC HTML below points to the apache2 default index page running on localhost, attached below, it contains no JS code and yet does not render in WeasyPrint.

[index.html.txt](https://github.com/Kozea/WeasyPrint/files/3979034/index.html.txt)

Test HTML 
 
```
<!doctype html>
<html>
<style>
body iframe {
   width: 720px;
   height: 720px;
}
</style>
<body>
<iframe src="http://localhost/index.html"><b>YYY</b></iframe>
</body>
</html>
```
Expected output (as rendered in browser):

![image](https://user-images.githubusercontent.com/6414463/71098018-e9442100-21a8-11ea-9f38-c2ff5c57c52d.png)

Actual output (as rendered in PDF):

![image](https://user-images.githubusercontent.com/6414463/71098120-1690cf00-21a9-11ea-95db-ae55606e23a3.png)

**Comment by liZe** [comment]
Wow.

I was sure that iframe were supported by WeasyPrint. I was so sure that I’ve already told this [in #840](https://github.com/Kozea/WeasyPrint/issues/840#issuecomment-478487626).

But it’s not. There’s no code about this. Not at all. We have `embed` and `object`, but not `iframe`. Sigh.

I don’t know how hard it would be to add this feature. Creating a new `Document`, rendering it and taking its first page to paint it in the current doccument seems possible, but I’m sure that there are many problems (the first one being the size of the rendered page).

**Comment by chutchut** [comment]
Yes, I also considered the depth of nested iframes to render as a potential issue (there would need to be a limit?), thanks for confirming though! :smile:

**Comment by liZe** [comment]
> Yes, I also considered the depth of nested iframes to render as a potential issue

We already have the problem with nested SVG images, and many more documented [security concerns](https://weasyprint.readthedocs.io/en/latest/tutorial.html#security).

**Comment by chutchut** [comment]
Hi @liZe any idea if/when this will be implemented?

**Comment by liZe** [comment]
Unfortunately, that’s probably not the text issue I’ll work on. But I can help someone who is interested in providing this feature!

**Comment by grewn0uille** [comment]
Hello!

If you’re interested in iframes, we created a [short survey](https://app.evalandgo.com/s/index.php?id=JTk4cCU5NG8lOUQlQUQ%3D&a=JTk3biU5M2glOUMlQUI%3D) where you can give a boost to this feature and help us to improve WeasyPrint 😉

Vote for it!
